src/riscv/lib/src/machine_state/memory/state.rs

```diff
@@ -55,8 +56,15 @@ impl<const PAGES: usize, const TOTAL_BYTES: usize, B, M: ManagerBase>
         B: Buddy<M>,
         M: ManagerReadWrite,
     {
-        self.protect_pages(0, TOTAL_BYTES, Permissions::READ_WRITE)
-            .unwrap();
+        use crate::machine_state::memory::DummyMemoryGovernanceHandler;
+
+        self.protect_pages(
+            0,
+            TOTAL_BYTES,
+            Permissions::READ_WRITE,
+            &mut DummyMemoryGovernanceHandler,
```

Comment:
Could this cause any problems with tests in the future? 

Should this method accept the listener parameter?

Or should it be moved higher in the hierarchy? For example, would it work in the `impl MachineState`?